Introduce `--no-import-callback` CLI option (#14610)
ethereum/solidity: docs/path-resolution.rst

> The command-line compiler provides the *Host Filesystem Loader* - a rudimentary callback
> that interprets a source unit name as a path in the local filesystem.
> The default callback can be disabled using the ``--no-import-callback`` command-line option, in which
> case, an error will be thrown when attempting to load sources that have not been explicitly provided.

Reviewer: How about this? The paragraph above already says that with no callback compilations fails.

```suggestion
By default, the command-line compiler provides the *Host Filesystem Loader* - a rudimentary callback
that interprets a source unit name as a path in the local filesystem.
This callback can be disabled using the ``--no-import-callback`` command-line option.
```

    The case must exactly match the one used in your imports.
    For example ``--allow-paths tokens`` will not match ``import "Tokens/IERC20.sol"``.

.. note::

>     Allowed paths is incompatible with the ``--no-import-callback`` option.

Reviewer: ```suggestion
    Specifying allowed paths is incompatible with the ``--no-import-callback`` option.
    The protection against loading files from unexpected locations is a feature
    of the Host Filesystem Loader, which the option disables.
```

But actually, does it even make sense to mention it? It's just a small detail that you'll find out right away if you try to use them together. I think it needs no explanation. Definitely nothing that stands out and draws attention, like a note.

Author: Sure, I'll remove it. I guess it's even implicitly implied in the above section.
